### ISSUE TITLE
Lab 8 Level 6 Test Case 

### ISSUE THREAD

**Issue by summerthia02** [nus-cs2030/2122-s2#312]
hii I'm stuck on this test case! does anyone know how to solve it? 
thanks in adv !

<img width="683" alt="Screenshot 2022-04-13 at 10 50 55 AM" src="https://user-images.githubusercontent.com/97536121/163090463-a95b877c-c54e-41c8-b90f-c567c6c64653.png">

**Comment by zuohui48** [comment]
When you take a sword from a room, the sword should be removed from that room. I think you can refer to #301. Hope it helps

**Comment by FanKJ13** [comment]
Yes when you take a sword, the boolean isTaken for the sword should be set to true and not be included in the output, unless you are present in the same room. Also, when you return to a room, you should check if the sword in the room is taken, if it's taken do not include it in your final list of Things.

**Comment by chloeoxe** [comment]
yup when u take a sword from one room to another, other than adding the sword to the new room's list of things, remember to remove the sword from the original room's list of things. It will probably be sufficient if u just implement this removing step for the go() method. For the back() method, u just need to check that the current room has a taken sword, and if it does, add it to the previous room, there is no need to remove the sword from the current room.

**Comment by Lu-Yi-Fan** [comment]
Hi! When you take a sword and move to another room, there should be a pairwise update. You remove the sword from the room that you are in and also add the sword to the room you are going to!
